packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+import Editor, { OnMount, useMonaco } from '@monaco-editor/react';
+import type { editor } from 'monaco-editor';
+import { DEFAULT_CHANGE_DEBOUNCE_TIME, DEFAULT_EDITOR_MODE, DEFAULT_EDITOR_THEME, DEFAULT_MONACO_OPTIONS } from './config';
+import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
+import { DebouncedFunc, debounce } from 'lodash-es';
+import { KvLoader } from '../stencil-generated';
+
+export const KvCodeEditor = ({
+	code,
+	readOnly = false,
+	debounceTime = DEFAULT_CHANGE_DEBOUNCE_TIME,
+	loadingComponent = <KvLoader />,
+	mode = DEFAULT_EDITOR_MODE,
+	theme = DEFAULT_EDITOR_THEME,
+	customTheme,
+	onChange
+}: ICodeEditorProps) => {
+	const monaco = useMonaco();
+	const editorRef = useRef<editor.IStandaloneCodeEditor>();
+	const readOnlyRef = useRef<boolean>(readOnly);
+
+	const onChangeDebounced: DebouncedFunc<OnEditorChangeCallback> = useMemo(() => debounce(value => onChange?.(value), debounceTime), []);
+	const onEditorMount: OnMount = useMemo(() => editor => (editorRef.current = editor), []);
+
+	useEffect(() => {
+		if (!monaco) return;
+		if (customTheme) {
```

Comment:
Would here make sense move this customTheme to the line above to clean up some code?

`if(!monaco || !customTheme) return;`

And then bellow without the if statement:
`
monaco.editor.defineTheme(ECodeEditorTheme.Custom, customTheme);
monaco.editor.setTheme(ECodeEditorTheme.Custom);
`

Reply:
Hmm I'd done this before but the linter would throw an error, I must've missed the not operator and got confused. Thanks for spotting it, I'll go ahead and change it :D